src/lib/comms/sol-socket.c

```diff
@@ -51,6 +51,7 @@ sol_socket_ip_new(const struct sol_socket_options *options)
 #ifdef DTLS
         s = sol_socket_default_dtls_new(options);
 #else
+        errno = ENOSYS;
         SOL_WRN("DTLS is not enabled, secure socket is not possible");
         return NULL;
```

Comment:
I am not checking for #ifdef DTLS in oic. I count on coap to check that.
Only using ifdefs in oic in security module. Is it better to check for DTLS in oic creation and don't even trying to create a secure coap connection? Mixed feelings about that.